File: DevEnvironment/Base/src/main/java/base/core/ResourcePathBuilder.java
package base.core;

/**
 * The ResourcePathBuilder class implements a builder of a resource path.
 */
public final class ResourcePathBuilder extends AbstractPathBuilder {
    private static final String INITIAL_COMPONENT = null;
    private static final String SEPARATOR = "/";

    /**
     * The ResourcePathBuilder constructor.
     */
    public ResourcePathBuilder() {
        super(
            ResourcePathBuilder.INITIAL_COMPONENT,
            ResourcePathBuilder.SEPARATOR);
    }
}

File: DevEnvironmentCLion/Base/core/ResourcePathBuilder.cpp
#include "PreCompiled.h"
#include "ResourcePathBuilder.h"

using namespace base;

const std::string ResourcePathBuilder::defaultRootDirectory;
const std::string ResourcePathBuilder::defaultSeparator = "/";

/**
 * The ResourcePathBuilder constructor.
 */
ResourcePathBuilder::ResourcePathBuilder(
    const std::string& rootDirectory,
    const std::string& separator) :
    AbstractPathBuilder(rootDirectory, separator)
{
}

/**
 * The ResourcePathBuilder Destructor.
 */
ResourcePathBuilder::~ResourcePathBuilder()
{
}